packages/orm/src/PluginManager.ts

```diff
@@ -35,23 +35,33 @@ interface PluginMethods {
    * @param rows The raw database rows returned from the query
    */
   afterFind?(meta: EntityMetadata, operation: FindOperation, rows: any[]): void;
-}
 
-const pluginMethods = ["beforeSetField", "beforeFind", "afterFind"] as (keyof PluginMethods)[];
+  afterWrite(entityTodos: Record<string, Todo>, joinRowTodos: Record<string, JoinRowTodo>): void;
+}
 
-const emSymbol = Symbol("em");
+const pluginMethods = ["beforeSetField", "beforeFind", "afterFind", "afterWrite"] as (keyof PluginMethods)[];
+const emsSymbol = Symbol("ems");
 /**
  * Base class for plugins that hook into entity lifecycle events.
  *
  * Plugins can implement any of the PluginMethods to intercept entity operations
  * and are automatically registered with the EntityManager when added via PluginManager.
  */
 export abstract class Plugin {
-  private [emSymbol]: EntityManager | undefined;
+  // We need to store the EntityManager references in a WeakRef array so that they can be garbage collected.  If we
+  // didn't do this, then multiple EMs with the same plugin instance could keep each other from being garbage collected.
+  [emsSymbol]: WeakRef<EntityManager>[] = [];
```

Comment:
Not against the weakref, but surely the GC would realize the "two EMs pointing at each other through a plugin" are a self-contained cycle, not reaching from any of the GC roots, and still collect them?